File: S/228-summary-ranges/main.go
package main

import (
	"fmt"
	"strconv"
)

func summaryRanges(nums []int) []string {
	if len(nums) == 0 {
		return nil
	}
	var ret []string
	start := -1
	prev := nums[0] - 2
	for i := 0; i < len(nums); i++ {
		if nums[i] != prev+1 {
			if i != 0 {
				if start == prev {
					ret = append(ret, strconv.FormatInt(int64(start), 10))
				} else {
					ret = append(ret, strconv.FormatInt(int64(start), 10)+"->"+strconv.FormatInt(int64(prev), 10))
				}
			}
			start = nums[i]
		}
		prev = nums[i]
	}
	if start == prev {
		ret = append(ret, strconv.FormatInt(int64(start), 10))
	} else {
		ret = append(ret, strconv.FormatInt(int64(start), 10)+"->"+strconv.FormatInt(int64(prev), 10))
	}
	return ret
}

func main() {
	fmt.Println("a")
}

File: S/228-summary-ranges/main.rs
struct Solution {}

impl Solution {
  pub fn summary_ranges(nums: Vec<i32>) -> Vec<String> {
    if nums.len() == 0 {
      return Vec::new();
    }
    let mut start: i32 = nums[0];
    let mut end: i32 = nums[0];
    let mut result: Vec<String> = Vec::new();
    (1..nums.len()).for_each(|i| {
      if nums[i] == end + 1 {
        end = nums[i];
      } else {
        if start == end {
          result.push(start.to_string());
        } else {
          result.push(format!("{}->{}", start, end));
        }
        start = nums[i];
        end = nums[i];
      }
    });
    if start == end {
      result.push(start.to_string());
    } else {
      result.push(format!("{}->{}", start, end));
    }
    result
  }
}
